robber/expect.py

```diff
@@ -32,14 +33,14 @@ def remove_custom_message(cls):
         cls.message = None
 
     @classmethod
-    def register(cls, name, klass):
+    def register(cls, name, klass, is_negative=False):
         cls.matchers[name] = klass
 
         def method(self, other=None, *args):
-            return klass(self.object, other, *args).fail_with(self.message).match()
+            fact = is_negative is not self.is_negative
```

Comment:
Let's take a look at how I registered the boolean matchers.

```python
expect.register('true', TrueMatcher)
expect.register('false', TrueMatcher, is_negative=True)
```

`self.is_negative` is the initial nature of the matcher. As for the TrueMatcher here, it it `False`.
`is_negative` is how we want to use it. As for the TrueMatcher registered as `false` here, it it `True`.
` is_negative != self.is_negative` decide the final nature of the matcher. I will put a fact table here:

| self.is_negative | is_negative | Fact  |
|------------------|-------------|-------|
| True             | True        | True  |
| True             | False       | False |
| False            | True        | False |
| False            | False       | True  |

From this fact table, I got that formula.